docs/developer-guide.md

```diff
@@ -0,0 +1,105 @@
+---
+title: Developer Guide
+toc: true
+weight: 5010
+---
+# Developer Guide
+Welcome to the Crossplane Developer Guide!
+
+## Overview 
```

Comment:
This document seems very specific to stacks, but as a reader, from the title I was expecting to learn more about developing Crossplane. Is this the Stacks developer guide? I would suggest that maybe we explain the high-level components of Crossplane, and which parts are covered by this guide, to help readers understand what they will be learning by reading this, and where they should go to learn more about other things.